src/java.naming/share/classes/com/sun/jndi/ldap/sasl/LdapSasl.java

```diff
@@ -243,5 +250,24 @@ private static NamingException wrapInNamingException(Exception e) {
         return ne;
     }
 
+    /**
+     * Parse value of "com.sun.jndi.ldap.tls.cbtype" property
+     * @param  cbType
+     * @return TLS Channel Binding type or null if
+     *         "com.sun.jndi.ldap.tls.cbtype" property has not been set.
+     * @throws ChannelBindingException
+     */
+    private static TlsChannelBindingType parseType(String cbType) throws ChannelBindingException {
```

Comment:
Maybe this method could throw NamingException directly now? That would avoid wrapping CBE into NamingException?

Reply:
My opinion is this method should be put back.